text/0039-_formatted-field-behavior.md

```diff
@@ -0,0 +1,136 @@
+            "_formatted": {
+                "title": "<em>Prince</em> Avalance"
+            }
+        }
+    ]
+}
+```
+
+but i get:
+```json
+{
+    "hits": [
+        {
+            "title": "Prince Avalanche",
+            "poster": "https://image.tmdb.org/t/p/w1280/3KHiQt54usbHyIjLIMzaDAoIJNK.jpg",
+            "_formatted": {
+                "title": "<em>Prince</em> Avalance",
+                "poster": "https://image.tmdb.org/t/p/w1280/3KHiQt54usbHyIjLIMzaDAoIJNK.jpg"
+            }
+        }
+    ]
+}
+```
+
+`_formatted` is filled with all the `displayedAttributes` despite the fact that the user only ask for one specific field.
+
+This behaviour will be fixed in the `0.21` release in order to only get the fields set in `attributesToHighlight` and `attributesToCrop` in `_formatted`.
+
+#### MeiliSearch Behavior (0.21)
+
+As in the `0.20`, the `_formatted` fields appears in the search response as soon as the `attributesToHighlight` and/or `attributesToCrop` parameters are sent with an empty value.
+
+E.g. `"attributesToHighlight": `
+
+As in the `0.20`, `_formatted` is also filled with all the `displayedAttributes` despite the fact that the user only ask for one specific field in `attributesForHighlight` or `attributesToCrop`.
+
+The `_formatted` field appears in the search response as soon as the `attributesToHighlight` and/or `attributesToCrop` parameters are sent as a query parameter and filled with an existent parameter.
+
+The next rules clarify the behavior of `_formatted` parameter given `attributesToHighlight` and `attributesToCrop` states and combination.
+
+✅  `_formatted` response parameter **SHOULD NOT** be returned in the given cases:
+
+- `attributesToHighlight` is set but is empty AND `attributesToCrop` is set but is empty.
+- `attributesToHighlight` is set but is empty AND `attributesToCrop` isn't set.
+- `attributesToHighlight` is set but only contains a non-existent field AND `attributesToCrop` is set but only contains a non-existent field.
+- `attributesToHighlight` is set but only contains a non-existent field AND (`attributesToCrop` is not set OR `attributesToCrop` is set but is empty).
+
+✅ `_formatted` response parameter **SHOULD** be returned in any cases of:
+
+- `attributesToHighlight` is set with `*`. When `attributesToHighlight` is not set, the engine will use `*` as default parameter. So each `displayedAttributes` will be in the `_formated` result in that specific case.
+- `attributesToCrop` is set with `*`.
+- `attributesToHighlight` is set with **1..N** existing attributes.
+- `attributesToCrop` is set with **1..N** existing attributes.
+
+
+> ⚠️ Note that if one of the two parameters is requested but empty or contains an attribute that does not exist but the other is valid, `_formatted` will be returned based on the valid parameter.
+
+> ⚠️ Note that `attributesToRetrieve` is independent of `_formatted` and so of `attributesToHighlight` and `attributesToCrop`.
```

Comment:
It seems to me that this is the easiest way for users to understand how to have what they decide to get as raw results and choose what they want to crop or highlight. Not mixing these parameters permits us to keep an easy way of describing the rules between each of these parameters. If the engine decides for them to mix these fields, we will need to explain each possible case and it seems complicated on so many levels. Moreover, they do not serve the same purpose.

@bidoubiwa and I have discussed that and that's why by default we want to highlight all the fields to keep an easy way for our users that do not read the documentation before using MeiliSearch to have a way to display and use results from `_formatted` field. @bidoubiwa explained to me that it will be easier for them to use MeiliSearch at first to directly get results from `_formatted` instead of merging some fields from `hits` and `formatted` by hand. By doing we allow keeping the engine an independent behavior between `attributesToRetrieve` and `_formatted` but we don't reduce the bootstrapping MeiliSearch experience.

For users that do not want to retrieve the whole possible fields, I've added a part in future possibilities to facilitate the removing of some fields instead of saying 'I want only this field and this field.. and so on' by using the `-fieldname` operator. E.g. If I want all the fields except one, I can write `*,-fieldname`. By removing highlighting on certain fields instead of typing all fields that the user may want,  it gives a better experience for the use case of there is a lot of fields in documents.

I'm happy to discuss that with you at any time for clarity!